apps/dashboard/src/@3rdweb-sdk/react/hooks/useContractRoles.ts

```diff
@@ -74,22 +74,27 @@ export function useIsAdminOrSelf<TContract extends ValidContractInstance>(
   return isAdmin;
 }
 
-export function useIsMinter<TContract extends ValidContractInstance>(
-  contract: RequiredParam<TContract>,
-) {
+export function useIsMinter(contract: ThirdwebContract) {
   const address = useActiveAccount()?.address;
-  const { data: contractType } = useContractType(contract?.getAddress());
-  const contractHasRoles = isContractWithRoles(contract);
-  const isAccountRole = useIsAccountRole(
-    "minter",
-    contractHasRoles ? contract : undefined,
-    address,
-  );
+  const { data: userCanMint, isError } = useReadContract(hasRole, {
+    contract,
+    targetAccountAddress: address ?? "",
+    role: "minter",
+    queryOptions: { enabled: !!address },
+  });
 
-  if (contractType === "custom") {
-    return true;
-  }
-  return isAccountRole;
+  const { data: anyoneCanMint, isError: isError2 } = useReadContract(hasRole, {
+    contract,
+    targetAccountAddress: ZERO_ADDRESS,
+    role: "minter",
+  });
```

Comment:
nitpick:

at this time point its easier to just _not_ destructure the objects and instead use them directly below:
- `userCanMintQuery` and `anyoneCanMintQuery`
- then access directly off of those (will make the code more readable)